src/utils/getCollectionsFilteredData.js

```diff
@@ -0,0 +1,4 @@
+export const getCollectionsFilteredData = (products, collections) => {
+  if (Object.values(collections).every((val) => val === false)) return products;
+  return products.filter((product) => collections[product.collectionId]);
+};
```

Comment:
Niceee you checked all products with good optimisation. Learnt something new from these lines.